packages/sample-app/sample_app/dataset_example.py

```diff
@@ -0,0 +1,302 @@
+    """Demonstrate updating a row in a dataset"""
+    print("\n=== Update Row Example ===")
+
+    try:
+        if not dataset.rows:
+            print("No rows to update")
+            return
+
+        # Get the first row
+        row = dataset.rows[0]
+        print(f"Updating row: {row.id}")
+
+        # Create update data
+        updates = {}
+        for column in dataset.columns:
+            if column.name == "product":
+                updates[column.id] = "Updated Product"
+            elif column.name == "price":
+                updates[column.id] = 28.0
+            elif column.name == "in_stock":
+                updates[column.id] = True
+            elif column.name == "category":
+                updates[column.id] = "Marketing"
+
+        # Update the row
+        row.update(updates)
+        updated_row = next((r for r in dataset.rows if r.id == row.id), None)
+        assert updated_row is not None
+        print(f"Updated row: ID={updated_row.id}, values={updated_row.values}")
+
+    except Exception as e:
+        print(f"Error updating row: {e}")
+
+
+def delete_row_example(dataset: Dataset):
+    """Demonstrate deleting a row from a dataset"""
+    print("\n=== Delete Row Example ===")
+
+    try:
+        num_rows = len(dataset.rows)
+        if not dataset.rows:
+            print("No rows to delete")
+            return
+
+        # Get the first row
+        row = dataset.rows[0]
+        row_id = row.id
+        print(f"Deleting row: {row_id}")
+
+        # Delete the row
+        row.delete()
+        print(f"Deleted row: {row_id}")
+
+        # Verify row is no longer in dataset
+        assert row_id not in [r.id for r in dataset.rows]
+        assert len(dataset.rows) == num_rows - 1
+
+    except Exception as e:
+        print(f"Error deleting row: {e}")
+
+
+def publish_dataset_example(dataset: Dataset) -> Optional[str]:
+    """Demonstrate publishing a dataset"""
+    print("\n=== Publish Dataset Example ===")
+
+    try:
+        published_version = dataset.publish()
+        print(f"Published dataset: {dataset.slug}, version: {published_version}")
+        return published_version
+
+    except Exception as e:
+        print(f"Error publishing dataset: {e}")
+        return None
+
+
+def get_dataset_by_version_example(slug: str, version: str):
+    """Demonstrate getting a dataset by version"""
+    print("\n=== Get Dataset by Version Example ===")
+
+    try:
+        dataset_csv = client.datasets.get_version_csv(slug=slug, version=version)
+        print(f"Retrieved dataset:\n{dataset_csv}")
+
+    except Exception as e:
+        print(f"Error getting dataset by version: {e}")
+
+
+def delete_dataset_example(slug: str):
+    """Demonstrate deleting a dataset"""
+    print("\n=== Delete Dataset Example ===")
+    try:
+        client.datasets.delete_by_slug(slug)
+        print("Dataset deleted")
+    except Exception as e:
+        print(f"Error deleting dataset: {e}")
+
+
+def main():
```

Comment:
would add some comment or ake some explanation what are you doing here?
make this example useful